geoh5py/objects/vp_model.py

```diff
@@ -0,0 +1,380 @@
+# ''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''
+#  Copyright (c) 2025 Mira Geoscience Ltd.                                     '
+#                                                                              '
+#  This file is part of geoh5py.                                               '
+#                                                                              '
+#  geoh5py is free software: you can redistribute it and/or modify             '
+#  it under the terms of the GNU Lesser General Public License as published by '
+#  the Free Software Foundation, either version 3 of the License, or           '
+#  (at your option) any later version.                                         '
+#                                                                              '
+#  geoh5py is distributed in the hope that it will be useful,                  '
+#  but WITHOUT ANY WARRANTY; without even the implied warranty of              '
+#  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the               '
+#  GNU Lesser General Public License for more details.                         '
+#                                                                              '
+#  You should have received a copy of the GNU Lesser General Public License    '
+#  along with geoh5py.  If not, see <https://www.gnu.org/licenses/>.           '
+# ''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''
+
+# pylint: disable=too-many-instance-attributes, too-many-arguments
+
+from __future__ import annotations
+
+import uuid
+from typing import cast
+
+import numpy as np
+
+from geoh5py.data import Data, PrimitiveTypeEnum, ReferencedData
+from geoh5py.objects import DrapeModel, Grid2D
+from geoh5py.objects.grid_object import GridObject
+from geoh5py.shared.utils import KEY_MAP, str2uuid, xy_rotation_matrix
+
+
+class VPModel(GridObject, DrapeModel):
+    """
+    VP Mesh object defined by a uniform grid in the u and v directions, and
+    draped layers and prisms that represent the vertical structure of the model.
+
+    Sub-class of :obj:`~geoh5py.objects.grid_object.GridObject` and
+    :obj:`~geoh5py.objects.drape_model.DrapeModel`.
+
+    :param u_cell_size: Cell size along the u-axis.
+    :param u_count: Number of cells along the u-axis.
+    :param v_cell_size: Cell size along the v-axis.
+    :param v_count: Number of cells along the v-axis.
+    :param flag_property_id: UUID or name of the flag property.
+    :param heterogeneous_property_id: UUID or name of the heterogeneous property.
+    :param physical_data_name: Name of the physical data.
+    :param unit_property_id: UUID or name of the unit property.
+    :param weight_property_id: UUID or name of the weight property.
+    """
+
+    _VALUE_MAP = {
```

Comment:
why naming it _VALUE_MAP? It's a bit confusing.
and why in maj, ist's a class object?
I thought MAJ were reserved to deafault instantiated objects in a file

Reply:
That's just a convention we used for static class attributes. We were initially using the dunder __VALUE_MAP, but it turns out that we can't overload the values of those on subclasses.